src/vsr/client_replies.zig

```diff
@@ -365,17 +394,14 @@ pub fn ClientRepliesType(comptime Storage: type) type {
                 client_replies.write_queue.push_assume_capacity(write);
                 client_replies.write_reply_next();
             }
+
+            assert(client_replies.writing.isSet(write.slot.index));
         }
 
         fn write_reply_next(client_replies: *ClientReplies) void {
             while (client_replies.write_queue.head()) |write| {
                 if (client_replies.writing.isSet(write.slot.index)) return;
 
-                var reads = client_replies.reads.iterate();
-                while (reads.next()) |read| {
-                    if (read.slot.index == write.slot.index) return;
-                }
-
```

Comment:
Maybe in `write_reply_callback` we could iterate the reads and assert that none of them are to the slot that we just wrote?

Reply:
There actually _can_ be races after the state sync.

TIL: in our testing storage, we forbid write-write races, but allow write-read races. Which ... seems fine? At least in this case.